### ISSUE TITLE
quoted-strings only-when-needed doesn't work properly for multiline strings

### ISSUE THREAD

**Issue by tjanez** [adrienverge/yamllint#275]
Example YAML file `input.yml`:

```yaml
---
multiline_url_with_quotes: "https://github.com/cloudposse/\
  github-authorized-keys/releases/download/1.0.3/\
  github-authorized-keys_linux_amd64"

multiline_url_without_quotes: https://github.com/cloudposse/\
  github-authorized-keys/releases/download/1.0.3/\
  github-authorized-keys_linux_amd64

```

Yamllint 1.23.0 reports:
```
$ yamllint -d '{extends: default, rules: {quoted-strings: {required: only-when-needed}}}' input.yml
input.yml
  2:28      error    string value is redundantly quoted with any quotes  (quoted-strings)

```

However, the quotes are actually not redundant:

```python
>>> import yaml
>>> with open("input.yml") as f:
...     data = yaml.safe_load(f)
... 
>>> data['multiline_url_with_quotes']
'https://github.com/cloudposse/github-authorized-keys/releases/download/1.0.3/github-authorized-keys_linux_amd64'
>>> data['multiline_url_without_quotes']
'https://github.com/cloudposse/\\ github-authorized-keys/releases/download/1.0.3/\\ github-authorized-keys_linux_amd64'
>>>
```

**Comment by adrienverge** [comment]
Thanks for the report! Contributions to fix this are welcome.

**Comment by julienfalque** [comment]
The issue also happens with values that would be parsed as octal values:

```yaml
foo: '0123456789'
```
